### ISSUE TITLE
CI failure:Failed to execute test case D00002/D00003, IPv6 gateway is not valid

### ISSUE THREAD

**Issue by iiiceoo** [spidernet-io/spiderpool#850]
**Output of the failure test**

```text
• [FAILED] [10.766 seconds]
test ippool CR [ippoolCR]
/home/runner/work/spiderpool/spiderpool/test/e2e/ippoolcr/ippoolcr_test.go:22
  [It] add a route with `routes` and `gateway` fields in the ippool spec [D00002, D00003, smoke, A00011]
  /home/runner/work/spiderpool/spiderpool/test/e2e/ippoolcr/ippoolcr_test.go:164

  Begin Captured GinkgoWriter Output >>
    create namespace ns3348-536159128 
    Succeeded to create ippool v4pool-3348-570615199 
    Succeeded to create ippool v6pool-3348-610654030 
    STEP: update v4 pool: invalid `gateway` and valid `route` 10/11/22 10:33:48.644
    STEP: update v4 pool: valid `gateway` and invalid `route` 10/11/22 10:33:48.685
    STEP: update v4 pool: valid `gateway` and `route` 10/11/22 10:33:48.714
    STEP: update v6 pool: invalid `gateway` and valid `route` 10/11/22 10:33:48.743
    STEP: update v6 pool: valid `gateway` and invalid `route` 10/11/22 10:33:48.773
    STEP: update v6 pool: valid `gateway` and `route` 10/11/22 10:33:48.8
    create pod ns3348-536159128/pod3348-644363592 
    wait for pod ns3348-536159128/pod3348-644363592 ready 
    pod ns3348-536159128/pod3348-644363592 ADDED event 

    pod ns3348-536159128/pod3348-644363592 status=Pending

    pod ns3348-536159128/pod3348-644363592 MODIFIED event 

    pod ns3348-536159128/pod3348-644363592 status=Pending

    pod ns3348-536159128/pod3348-644363592 MODIFIED event 

    pod ns3348-536159128/pod3348-644363592 status=Pending

    pod ns3348-536159128/pod3348-644363592 MODIFIED event 

    pod ns3348-536159128/pod3348-644363592 status=Pending

    pod ns3348-536159128/pod3348-644363592 MODIFIED event 

    pod ns3348-536159128/pod3348-644363592 status=Pending

    pod ns3348-536159128/pod3348-644363592 MODIFIED event 

    pod ns3348-536159128/pod3348-644363592 status=Running

    pod: ns3348-536159128/pod3348-644363592, ips: [{IP:10.252.193.2} {IP:fd00:127::3}] 
    succeeded to check pod ipv4 ip
    succeeded to check pod ipv6 ip
    create pod ns3348-536159128/pod3348-644380792 
    wait for pod ns3348-536159128/pod3348-644380792 ready 
    pod ns3348-536159128/pod3348-644380792 ADDED event 

    pod ns3348-536159128/pod3348-644380792 status=Pending

    pod ns3348-536159128/pod3348-644380792 MODIFIED event 

    pod ns3348-536159128/pod3348-644380792 status=Pending

    pod ns3348-536159128/pod3348-644380792 MODIFIED event 

    pod ns3348-536159128/pod3348-644380792 status=Pending

    pod ns3348-536159128/pod3348-644380792 MODIFIED event 

    pod ns3348-536159128/pod3348-644380792 status=Pending

    pod ns3348-536159128/pod3348-644380792 MODIFIED event 

    pod ns3348-536159128/pod3348-644380792 status=Pending

    pod ns3348-536159128/pod3348-644380792 MODIFIED event 

    pod ns3348-536159128/pod3348-644380792 status=Running

    pod: ns3348-536159128/pod3348-644380792, ips: [{IP:10.252.193.3} {IP:fd00:127::6}] 
    succeeded to check pod ipv4 ip
    succeeded to check pod ipv6 ip
    check whether the route information is effective
    delete namespace ns3348-536159128 
    Delete IPv4 pools v4pool-3348-570615199 and IPv6 pools v6pool-3348-610654030 
  << End Captured GinkgoWriter Output

  Expected
      <[]uint8 | len:0, cap:512>: 
  not to be empty
  In [It] at: /home/runner/work/spiderpool/spiderpool/test/e2e/ippoolcr/ippoolcr_test.go:324
```

**Log file**
[e2edebugLog.txt](https://github.com/spidernet-io/spiderpool/files/9761920/e2edebugLog.txt)

**Link of the CI build**
https://github.com/spidernet-io/spiderpool/actions/runs/3226021852/jobs/5279105064

**Additional**

This morning, the CI in one of my PRs reported that the IPv6 gateway did not take effect in the container. I think it's strange, because this PR only adds a test to New function of Limiter, and nothing else has been changed, so I don't think my PR have introduced any new errors.

But here is an incorrect assertion:

![xxx_16655524902764](https://user-images.githubusercontent.com/53129907/195267721-1c99de79-dd2e-40e2-93f5-8ff93a894666.png)

The codes are as follows:

![xxx_166555252152](https://user-images.githubusercontent.com/53129907/195267650-364931f6-c8c3-4c52-b280-8e118fd45cd4.png)

That is, executing `ip -6 r | grep 'default via {IPV6_GATEWAY}'` in the container does not retrieve anything.

I checked the relevant code, but found no problem. Then I found the IPPool that set the gateway through the log I downloaded:

![xxx_16655527246708](https://user-images.githubusercontent.com/53129907/195267543-9e7854d9-e2e9-45c9-80d9-8bcfe4127584.png)

And, the call result of IPAM CNI of a Pod that allocated IP addresses through the IPPool contains the default  IPv6 route:

![xxx_16655527002218](https://user-images.githubusercontent.com/53129907/195267497-99ce3943-f6f3-449b-be1e-45a75abb7770.png)

Therefore, I think the result returned by IPAM CNI to Main CNI should be correct, and it is doubtful whether Main CNI takes effect on this default route. So I created an identical IPv6 IPPool, and then asked a Pod to allocate IP addresses through it. As a result, I saw both general and gateway generated routes in the container:

![xxx_16655529556892](https://user-images.githubusercontent.com/53129907/195267421-04b47a11-a10e-4249-8a9a-3f836c6a3d45.png)

Back to the beginning, why did this assertion fail?

**Comment by ty-dc** [comment]
**The reason is known and I will fix it as soon as possible.**

The reasons for this are as follows: 
- For ipv6 scenarios, the format of the PodRouteValue when added is as follows: 
```
fd00:015d::1
```
- When the application is created, it actually takes effect as:
```
fd00:15d::1
```
- The check command in the case is: 
```
ip -6 r | grep 'default via fd00:015d::1
```
**In this case, the content cannot be retrieved using the above command, so the os/exec execution returns an error**

**Comment by ty-dc** [comment]
fix #893
